File: back/back/Models/PlaneteOrigine.cs
using LiteDB;

namespace back.Models;

public sealed class PlaneteOrigine
{
     [BsonId]
    public int Id { get; set; }

     [BsonRef]
     public Systeme Systeme { get; set; } = null!;

     public string Nom { get; set; } = null!;
     public string? Description { get; set; }
     public string? NomFichier { get; set; }

     /// <summary>
     /// Statut de l'astre (Vitrifié, en guerre...)
     /// </summary>
     public int Statut { get; set; }

     /// <summary>
     /// A qui appartient l'astre (UNSC, covenante...)
     /// </summary>
     public int Appartenance { get; set; }

     /// <summary>
     /// Type d'astre
     /// Lune, planete, asteroide, etoile
     /// </summary>
     public int Type { get; set; }

     public int Densite { get; set; }
     public bool EstPlaneteOrigine { get; set; }

    public decimal PositionX { get; set; }
    public decimal PositionY { get; set; }

     public List<Orbite> ListeOrbite { get; set; } = [];
}

public sealed class Orbite
{
     public decimal OrbiteDecalageX { get; set; }
     public decimal OrbiteDecalageY { get; set; }
     public decimal OrbiteX { get; set; }
     public decimal OrbiteY { get; set; }
     public decimal OrbiteAngle { get; set; }
}

File: front/src/models/PlaneteOrigine.ts
import { EAppartenancePlanete, EStatusPlanete, ETypePlanete } from "@enums/EStatusPlanete"

export type PlaneteOrigine =
{
    id: number,
    idSysteme: number,
    nom: string,
    description: string | null,
    nomFichier: string,
    estPlaneteOrigine: boolean,
    statut: EStatusPlanete,
    type: ETypePlanete,
    appartenance: EAppartenancePlanete,
    densite: number,
    
    listeOrbite: Orbite[],
    
    positionX: number,
    positionY: number
}

export type Orbite = 
{
    orbiteDecalageX: number | null,
    orbiteDecalageY: number | null,
    orbiteX: number | null,
    orbiteY: number | null,
    orbiteAngle: number | null,
}

export type PlaneteConnecter = 
{
    idPlaneteA: number,
    idPlaneteB: number,
    distance: string | null
}

export type PlaneteOrigineLeger =
{
    id: number,
    nom: string
}

export type PlaneteOrigineRequete = 
{
    nom: string,
    description: string | null,
    statut: EStatusPlanete,
    estPlaneteOrigine: boolean,
    type: number,
    appartenance: number,

    positionX: number,
    positionY: number
}

export type PlaneteConnecterSupprimerRequete = Omit<PlaneteConnecter, "distance">;